File: lua/ispf.lua
-- Copyright 2026 drblez AKA Ruslan Stepanenko (drblez@gmail.com)
--
-- Lua/TSO ISPF module stubs.
--
-- Object Table:
-- | Object | Kind | Purpose |
-- |--------|------|---------|
-- | ispf.qry | function | Query ISPF environment |
-- | ispf.exec | function | Execute ISPF command |
-- | ispf.vget | function | Get ISPF variables |
-- | ispf.vput | function | Put ISPF variables |
local ispf = {}

function ispf.qry()
  error("LUZ12001 ispf.qry not implemented")
end

function ispf.exec(_)
  error("LUZ12002 ispf.exec not implemented")
end

function ispf.vget(_,_)
  error("LUZ12003 ispf.vget not implemented")
end

function ispf.vput(_,_)
  error("LUZ12004 ispf.vput not implemented")
end

return ispf

File: src/ispf.c
/*
 * Copyright 2026 drblez AKA Ruslan Stepanenko (drblez@gmail.com)
 *
 * Lua/TSO ISPF API stubs.
 *
 * Object Table:
 * | Object | Kind | Purpose |
 * |--------|------|---------|
 * | lua_ispf_qry | function | Query ISPF environment |
 * | lua_ispf_exec | function | Execute ISPF command |
 */
#include "ISPF"
#include "ERRORS"

int lua_ispf_qry(void)
{
  return LUZ_E_ISPF_QRY;
}

int lua_ispf_exec(const char *cmdline)
{
  (void)cmdline;
  return LUZ_E_ISPF_EXEC;
}
